docs/en/sql-reference/functions/ext-dict-functions.md

```diff
@@ -481,44 +443,1954 @@ SELECT dictGetAll('regexp_dict', 'tag', 'foobarbaz', 2);
-- `default_value_expr` — Value returned if the dictionary does not contain a row with the `id_expr` key. [Expression](../../sql-reference/syntax.md#syntax-expressions) returning the value in the data type configured for the `attr_name` attribute.
+- `dictionary_name`: the name of the dictionary you want to retrieve data from. [String literal](../syntax#syntax-string-literal)
+- `attribute_name`: the attribute whose value you want to get. [String literal](../syntax#syntax-string-literal), or [tuple](../data-types/tuple) of column names.
+- `key`: the key associated with the attribute. [Expression](../syntax#syntax-expressions) returning dictionary key-type value.
+- `default_value_expr` — Values returned if the dictionary does not contain a row with the `id_expr` key. [Expression](../../sql-reference/syntax.md#syntax-expressions) or [Tuple](../../sql-reference/data-types/tuple.md)([Expression](../../sql-reference/syntax.md#syntax-expressions)), returning the value (or values) in the data types configured for the `attr_names` attribute.
 
 **Returned value**
 
-- If ClickHouse parses the attribute successfully in the [attribute’s data type](../../sql-reference/dictionaries/index.md#dictionary-key-and-fields#ext_dict_structure-attributes), functions return the value of the dictionary attribute that corresponds to `id_expr`.
+The value of the dictionary attribute that corresponds to the given key. This value is parsed and returned as an `Int8` type. If the attribute cannot be found, the value of `default_value_expr` is returned instead.
 
-- If there is no requested `id_expr` in the dictionary then:
+**Implementation details**
 
-        - `dictGet[Type]` returns the content of the `<null_value>` element specified for the attribute in the dictionary configuration.
-        - `dictGet[Type]OrDefault` returns the value passed as the `default_value_expr` parameter.
+This function will throw an exception if the key does not exist in the dictionary.
 
-ClickHouse throws an exception if it cannot parse the value of the attribute or the value does not match the attribute data type.
+**Examples**
+
+Query:
+
+```sql
+CREATE TABLE countries
+(
+    id UInt64,
+    country String,
+    code_designation Int8
+) ENGINE = MergeTree
+ORDER BY id;
+
+INSERT INTO countries VALUES (1, 'Mali', 32);
+INSERT INTO countries VALUES (2, 'Czechia', 28);
+INSERT INTO countries VALUES (3, 'Brazil', 82);
+
+CREATE DICTIONARY country_codes
+(
+    id UInt64,
+    country String,
+    code_designation Int8
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE 'countries'))
+LAYOUT(FLAT())
+LIFETIME(MIN 0 MAX 1000);
+
+SELECT dictGetInt8OrDefault('country_codes', 'code_designation', '999', 'Default return string.') AS code_designation_output;
```

Comment:
This query will throw an exception that it cannot parse `'Default return string.'` as `Int8`

Reply:
```suggestion
SELECT dictGetInt8OrDefault('country_codes', 'code_designation', '999', 42) AS code_designation_output;
```

And the same for all other `dictGet<type>OrDefault`